### PR TITLE
Spoke

### PR DESCRIPTION
**Pull request by derpy-duck**

### CODE REVIEW

File: example-2/evm/src/contracts/lendingSpoke/Spoke.sol [hunk 1]
```diff
@@ -4,33 +4,108 @@ pragma solidity ^0.8.0;
...
+        DepositPayload memory depositPayload = DepositPayload({
+            header: payloadHeader,
+            assetAddress: assetAddress,
+            assetAmount: assetAmount
+        });
+
+        // create WH message
+        bytes memory serialized = encodeDepositPayload(depositPayload);
+
+        sendTokenBridgeMessage(assetAddress, assetAmount, serialized);
```
Comment (wrinkledLatherer) [inline example-2/evm/src/contracts/lendingSpoke/Spoke.sol:53]: you want to do an actual token transfer here too, where the user sends tokens to the protocol pool

File: example-2/evm/src/contracts/lendingSpoke/Spoke.sol [hunk 2]
```diff
@@ -4,33 +4,108 @@ pragma solidity ^0.8.0;
...
+        RepayPayload memory repayPayload = RepayPayload({
+            header: payloadHeader,
+            assetAddress: assetAddress,
+            assetAmount: assetAmount
+        });
+
+        // create WH message
+        bytes memory serialized = encodeRepayPayload(repayPayload);
+
+        sendTokenBridgeMessage(assetAddress, assetAmount, serialized);
```
Comment (wrinkledLatherer) [inline example-2/evm/src/contracts/lendingSpoke/Spoke.sol:107]: again, want to send a bunch of tokens to the protocol. may not want to do multiple token mints for the reason that batch VAAs aren't fully out yet, but even for one token we want to send tokens to protocol before send WH msg

File: example-2/evm/src/contracts/lendingSpoke/SpokeState.sol [hunk 3]
```diff
@@ -1,7 +1,9 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8.0;
 
-contract SpokeStorage {
+import "../lendingHub/HubStructs.sol";
+
+contract SpokeStorage is HubStructs {
```
Comment (wrinkledLatherer) [inline example-2/evm/src/contracts/lendingSpoke/SpokeState.sol:6]: why SpokeStorage instead of SpokeState?

Comment (derpy-duck) [reply example-2/evm/src/contracts/lendingSpoke/SpokeState.sol:6]: that's a great question but I was just following the convention in Hub:
`contract HubStorage is HubStructs {
    struct Provider {
        uint16 chainId;
        address payable wormhole;
        address tokenBridge;
        address pyth;
    }

    struct State {
        Provider provider;
        `

**Pull request merged**